core/src/main/java/com/klaytn/caver/tx/gas/DefaultGasProvider.java

```diff
@@ -21,25 +21,72 @@
 package com.klaytn.caver.tx.gas;
 
 import com.klaytn.caver.Caver;
+import com.klaytn.caver.methods.response.BlockHeader;
+import com.klaytn.caver.methods.response.Quantity;
 import com.klaytn.caver.tx.ManagedTransaction;
 import com.klaytn.caver.tx.SmartContract;
-import org.web3j.tx.gas.StaticGasProvider;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.tx.gas.ContractGasProvider;
 
+import java.io.IOException;
 import java.math.BigInteger;
 
 /**
  * @deprecated This class is deprecated since caver-java:1.5.0
  */
 @Deprecated
```

Comment:
Is this fine to put here?

Reply:
I think so. That exception can be triggered when DefaultGasProvider try to fetch gas price by Klaytn Node. (e.g. connected node does not work or given url at Caver instance is wrong, etc...)